src/components/FilterSuggester/FilterSuggester.jsx

```diff
@@ -0,0 +1,306 @@
+      { name: 'comments', prefix: 'sort:' },
+      { name: 'reactions', prefix: 'sort:' },
+      { name: 'interactions', prefix: 'sort:' },
+      { name: 'reactions-+1-desc', prefix: 'sort:' },
+      { name: 'reactions--1-desc', prefix: 'sort:' },
+      { name: 'reactions-smile-desc', prefix: 'sort:' },
+      { name: 'reactions-thinking_face-desc', prefix: 'sort:' },
+      { name: 'reactions-heart-desc', prefix: 'sort:' },
+      { name: 'reactions-tada-desc', prefix: 'sort:' },
+      { name: 'reactions-+1', prefix: 'sort:' },
+      { name: 'reactions--1', prefix: 'sort:' },
+      { name: 'reactions-smile', prefix: 'sort:' },
+      { name: 'reactions-thinking_face', prefix: 'sort:' },
+      { name: 'reactions-heart', prefix: 'sort:' },
+      { name: 'reactions-tada', prefix: 'sort:' },
+    ],
+  },
+  { name: 'milestone:', tip: 'milestone name' },
+  { name: 'author:', tip: 'username' },
+  { name: 'assignee:', tip: 'username' },
+  { name: 'mentions:', tip: 'username' },
+  { name: 'commenter:', tip: 'username' },
+  { name: 'involves:', tip: 'username' },
+  { name: 'team:', tip: 'team name' },
+  { name: 'user:', tip: 'username' },
+  { name: 'org:', tip: 'organization name' },
+  { name: 'repo:', tip: 'user/repo or org/repo' },
+  { name: 'created:', tip: 'date' },
+  { name: 'updated:', tip: 'date' },
+  { name: 'closed:', tip: 'date' },
+  { name: 'merged:', tip: 'date' },
+  { name: 'head:', tip: 'branch name' },
+  { name: 'base:', tip: 'branch name' },
+  {
+    name: 'comments:',
+    tip: 'number',
+    suggestions: [
+      { name: '', prefix: 'comments:', tip: 'number' },
+      { name: '>', prefix: 'comments:', tip: 'number' },
+      { name: '<', prefix: 'comments:', tip: 'number' },
+      { name: '>=', prefix: 'comments:', tip: 'number' },
+      { name: '<=', prefix: 'comments:', tip: 'number' },
+    ],
+  },
+  {
+    name: 'reactions:',
+    tip: 'number',
+    suggestions: [
+      { name: '', prefix: 'reactions:', tip: 'number' },
+      { name: '>', prefix: 'reactions:', tip: 'number' },
+      { name: '<', prefix: 'reactions:', tip: 'number' },
+      { name: '>=', prefix: 'reactions:', tip: 'number' },
+      { name: '<=', prefix: 'reactions:', tip: 'number' },
+    ],
+  },
+  {
+    name: 'interactions:',
+    tip: 'number',
+    suggestions: [
+      { name: '', prefix: 'interactions:', tip: 'number' },
+      { name: '>', prefix: 'interactions:', tip: 'number' },
+      { name: '<', prefix: 'interactions:', tip: 'number' },
+      { name: '>=', prefix: 'interactions:', tip: 'number' },
+      { name: '<=', prefix: 'interactions:', tip: 'number' },
+    ],
+  },
+]
+
+class FilterSuggester extends React.Component {
+  constructor(props) {
+    super(props)
+    const value = props.value || ''
+    this.state = {
+      value,
+      previousValue: value,
+      suggestions: [],
+    }
+    this.onKeyPress = this.onKeyPress.bind(this)
+    this.onKeyUp = this.onKeyUp.bind(this)
+  }
```

Comment:
Can you explain a little about what this `constructor` does? Does it just rebind the keypresses and reset the value so the autocomplete can change every time the user types a letter?